dev/AppNotifications/AppNotificationManager.cpp

```diff
@@ -132,6 +132,47 @@ namespace winrt::Microsoft::Windows::AppNotifications::implementation
         m_notificationHandlers.remove(token);
     }
 
+    HRESULT __stdcall AppNotificationManager::Activate(
+        LPCWSTR /* appUserModelId */,
+        LPCWSTR invokedArgs,
+        [[maybe_unused]] NOTIFICATION_USER_INPUT_DATA const* data,
+        [[maybe_unused]] ULONG dataCount) noexcept try
+    {
+        winrt::IMap<winrt::hstring, winrt::hstring> userInput{ winrt::single_threaded_map<winrt::hstring, winrt::hstring>() };
+        for (unsigned long i = 0; i < dataCount; i++)
+        {
+            userInput.Insert(data[i].Key, data[i].Value);
+        }
+
+        winrt::AppNotificationActivatedEventArgs activatedEventArgs = winrt::make<winrt::Microsoft::Windows::AppNotifications::implementation::AppNotificationActivatedEventArgs>(invokedArgs, userInput);
+
+        // Need to store the first notification in the case of ToastActivation
+        if (!m_firstNotificationReceived)
```

Comment:
Lock Lock Lock!

Reply:
Isn't this already synchronized by ComActivation?

Reply:
Otherwise 2 activations can race through the if condition and cause problems. This if is in essence atomic.

Reply:
We can have two threads that call CoCreateInstance at the same time and the processor can context switch!